### PR TITLE
refactor: drop decorators

### PR DESCRIPTION
**Pull request by splincode**
Close https://github.com/taiga-family/taiga-ui/issues/8346

image

### CODE REVIEW

File: projects/addon-mobile/components/pull-to-refresh/loader-android/loader-android.component.ts [hunk 1]
```diff
@@ -18,25 +18,26 @@ const ROTATE_X_MULTIPLIER = 2.3;
     templateUrl: './loader-android.template.html',
     styleUrls: ['./loader-android.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
+    host: {
+        '[class._visible]': 'dropped',
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/pull-to-refresh/loader-android/loader-android.component.ts:22]: Probably, it is a mistake

File: projects/addon-mobile/components/sheet-dialog/sheet-dialog.component.ts [hunk 2]
```diff
@@ -109,10 +108,15 @@ export class TuiSheetDialogComponent<I> implements AfterViewInit {
...
     }
 
-    @HostListener('document:touchstart.passive.silent', ['1'])
-    @HostListener('document:touchend.silent', ['-1'])
-    @HostListener('document:touchcancel.silent', ['-1'])
-    @HostListener('scroll.silent', ['0'])
+    @shouldCall(function (this: TuiSheetDialogComponent<unknown>): boolean {
+        return this.context.closeable;
+    })
+    protected close(): void {
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/sheet-dialog/sheet-dialog.component.ts:114]: Looks strange 🤔

File: projects/addon-table/components/reorder/reorder.component.ts [hunk 3]
```diff
@@ -33,6 +32,11 @@ import {TUI_REORDER_OPTIONS} from './reorder.options';
     templateUrl: './reorder.template.html',
     styleUrls: ['./reorder.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
+    host: {
+        '(focusout.stop)': 'noop()',
```
Comment (nsbarsukov) [inline projects/addon-table/components/reorder/reorder.component.ts:36]:
```suggestion
        '(focusout.stop)': '0',
```

File: projects/core/components/data-list/data-list.component.ts [hunk 4]
```diff
@@ -44,6 +42,15 @@ export function tuiInjectDataListSize(): TuiSizeL | TuiSizeS {
     providers: [tuiAsDataListAccessor(TuiDataListComponent)],
     host: {
         role: 'listbox',
+        '[attr.data-size]': 'size',
+        '(focusin)': 'onFocusIn($event.relatedTarget, $event.currentTarget)',
+        '(mousedown.prevent)': 'noop()',
```
Comment (nsbarsukov) [inline projects/core/components/data-list/data-list.component.ts:47]:
```suggestion
        '(mousedown.prevent)': '0',
```

File: projects/core/components/data-list/data-list.component.ts [hunk 5]
```diff
@@ -44,6 +42,15 @@ export function tuiInjectDataListSize(): TuiSizeL | TuiSizeS {
...
+        '(wheel.silent.passive)': 'handleFocusLossIfNecessary()',
+        '(mouseleave)': 'handleFocusLossIfNecessary($event.target)',
+        '(keydown.tab)': 'handleFocusLossIfNecessary($event.target)',
+        '(keydown.shift.tab)': 'handleFocusLossIfNecessary($event.target)',
```
Comment (nsbarsukov) [inline projects/core/components/data-list/data-list.component.ts:51]: According to the previous version, `$event.target` is required only for `mouseleave`

File: projects/core/components/expand/expand.component.ts [hunk 6]
```diff
@@ -47,6 +45,13 @@ export const TUI_EXPAND_LOADED = 'tui-expand-loaded';
     animations: [tuiParentAnimation],
     host: {
         ngSkipHydration: 'true',
+        '[style.height.px]': 'height',
+        '[class._loading]': 'loading',
+        '[class._overflow]': 'overflow',
+        '[class._expanded]': 'expanded',
+        '[attr.aria-expanded]': 'expanded',
+        '(tui-expand-loaded)': 'onExpandLoaded($event)',
```
Comment (nsbarsukov) [inline projects/core/components/expand/expand.component.ts:53]: Could we use const `TUI_EXPAND_LOADED` ?

File: projects/core/components/scrollbar/scrollbar.component.ts [hunk 7]
```diff
@@ -38,7 +37,11 @@ export const TUI_SCROLLABLE = 'tui-scrollable';
             useFactory: () => inject(TuiScrollbar).browserScrollRef,
         },
     ],
-    host: {'[class._native-hidden]': '!isIOS || hidden'},
+    host: {
+        '[class._native-hidden]': '!isIOS || hidden',
+        '(tui-scrollable.stop)': 'onScrollable($event.detail)',
+        '(tui-scroll-into-view.stop)': 'scrollIntoView($event.detail)',
```
Comment (nsbarsukov) [inline projects/core/components/scrollbar/scrollbar.component.ts:43]: What's about constants `TUI_SCROLLABLE` / `TUI_SCROLL_INTO_VIEW` ?

File: projects/core/directives/dropdown/dropdown-context.directive.ts [hunk 8]
```diff
@@ -46,40 +52,28 @@ export class TuiDropdownContext extends TuiRectAccessor {
         return this.currentRect;
     }
 
-    @HostListener('contextmenu.prevent.stop', ['$event.clientX', '$event.clientY'])
+    @shouldCall(function (this: TuiDropdownContext, target: Element): boolean {
+        return !this.activeZone.contains(target);
+    })
```
Comment (nsbarsukov) [inline projects/core/directives/dropdown/dropdown-context.directive.ts:57]: Why so ?

File: projects/addon-mobile/components/pull-to-refresh/loader-android/loader-android.component.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-mobile/components/pull-to-refresh/loader-android/loader-android.component.ts]: fixed

File: projects/core/directives/dropdown/dropdown-open.directive.ts [hunk 9]
```diff
@@ -105,15 +101,26 @@ export class TuiDropdownOpen implements OnChanges {
...
-    @HostListener('click', ['$event.target'])
+    @shouldCall(function (this: TuiDropdownOpen, event: Event | KeyboardEvent): boolean {
+        return (
+            'key' in event &&
+            event.key.toLowerCase() === 'escape' &&
+            this.tuiDropdownEnabled &&
+            !!this.tuiDropdownOpen &&
+            !this['dropdown']()?.nextElementSibling
+        );
+    })
```
Comment (nsbarsukov) [inline projects/core/directives/dropdown/dropdown-open.directive.ts:112]: why ?

File: projects/kit/components/carousel/carousel.component.ts [hunk 10]
```diff
@@ -52,6 +50,13 @@ import {TuiCarouselScroll} from './carousel-scroll.directive';
             inputs: ['duration'],
         },
     ],
+    host: {
+        '[class._transitioned]': 'draggable',
```
Comment (nsbarsukov) [inline projects/kit/components/carousel/carousel.component.ts:54]: Probably, it is a mistake

Comment (nsbarsukov) [reply projects/kit/components/carousel/carousel.component.ts:54]: + missing `class._draggable`

File: projects/kit/components/tabs/tabs.directive.ts [hunk 11]
```diff
@@ -34,7 +32,10 @@ class TuiTabsStyles {}
 
 @Directive({
     standalone: true,
-    host: {'[attr.data-size]': 'size'},
+    host: {
+        '[attr.data-size]': 'size',
+        '(tui-tab-activate)': 'onActivate($event, $event.target)',
```
Comment (nsbarsukov) [inline projects/kit/components/tabs/tabs.directive.ts:37]: Constant `TUI_TAB_ACTIVATE` ?

File: projects/kit/components/tiles/tile-handle.directive.ts [hunk 12]
```diff
@@ -47,9 +31,21 @@ export class TuiTileHandle {
         this.tile.onDrag([NaN, NaN]);
     }
 
-    @shouldCall(isDragging)
-    @HostListener('document:pointermove.silent', ['$event.x', '$event.y'])
+    @shouldCall(function (this: TuiTileHandle): boolean {
+        return !Number.isNaN(this['x']);
+    })
```
Comment (nsbarsukov) [inline projects/kit/components/tiles/tile-handle.directive.ts:36]: I am not sure that we actually need this change

File: projects/legacy/components/sheet/components/sheet/sheet.component.ts [hunk 13]
```diff
@@ -36,6 +34,7 @@ import {TUI_SHEET_PROVIDERS} from './sheet.providers';
         role: 'dialog',
         '[attr.aria-labelledby]': 'id',
         '[class._ios]': 'isIos',
+        '(tui-sheet-id)': 'onId($event.detail)',
```
Comment (nsbarsukov) [inline projects/legacy/components/sheet/components/sheet/sheet.component.ts:37]: constant `TUI_SHEET_ID` ?

File: projects/addon-mobile/components/sheet-dialog/sheet-dialog.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-mobile/components/sheet-dialog/sheet-dialog.component.ts]: fixed

File: projects/addon-table/components/reorder/reorder.component.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-table/components/reorder/reorder.component.ts]: fixed

File: projects/core/components/data-list/data-list.component.ts (hunk 4 above)
Comment (splincode) [reply projects/core/components/data-list/data-list.component.ts]: fixed

File: projects/core/components/data-list/data-list.component.ts (hunk 5 above)
Comment (splincode) [reply projects/core/components/data-list/data-list.component.ts]: fixed

File: projects/core/components/expand/expand.component.ts (hunk 6 above)
Comment (splincode) [reply projects/core/components/expand/expand.component.ts]: fixed

File: projects/core/components/scrollbar/scrollbar.component.ts (hunk 7 above)
Comment (splincode) [reply projects/core/components/scrollbar/scrollbar.component.ts]: fixed

File: projects/core/directives/dropdown/dropdown-context.directive.ts (hunk 8 above)
Comment (splincode) [reply projects/core/directives/dropdown/dropdown-context.directive.ts]: fixed

File: projects/core/directives/dropdown/dropdown-open.directive.ts (hunk 9 above)
Comment (splincode) [reply projects/core/directives/dropdown/dropdown-open.directive.ts]: fixed

File: projects/kit/components/carousel/carousel.component.ts (hunk 10 above)
Comment (splincode) [reply projects/kit/components/carousel/carousel.component.ts]: fixed

File: projects/kit/components/tabs/tabs.directive.ts (hunk 11 above)
Comment (splincode) [reply projects/kit/components/tabs/tabs.directive.ts]: fixed

File: projects/kit/components/tiles/tile-handle.directive.ts (hunk 12 above)
Comment (splincode) [reply projects/kit/components/tiles/tile-handle.directive.ts]: fixed

File: projects/legacy/components/sheet/components/sheet/sheet.component.ts (hunk 13 above)
Comment (splincode) [reply projects/legacy/components/sheet/components/sheet/sheet.component.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Review by waterplea — COMMENTED** [review]
Would be nice to sort them so class/style/attr are all next to each other

**Pull request merged**